lib/src/components/date_input/date_input.dart

```diff
@@ -1,319 +1,314 @@
+
+  double get _iconSize {
+    switch (widget.size) {
+      case ZetaWidgetSize.large:
+        return ZetaSpacing.x6;
+      case ZetaWidgetSize.medium:
+        return ZetaSpacing.x5;
+      case ZetaWidgetSize.small:
+        return ZetaSpacing.x4;
+    }
+  }
+
+  DateTime? get _value {
+    final value = _dateFormatter.getMaskedText().trim();
+    final date = DateFormat(widget.dateFormat).tryParseStrict(value);
+
+    return date;
+  }
 
   @override
   void initState() {
-    super.initState();
-    _hintText = widget.datePattern.toLowerCase();
     _dateFormatter = MaskTextInputFormatter(
-      mask: _hintText.replaceAll(RegExp('[a-z]'), '#'),
+      mask: widget.dateFormat.replaceAll(RegExp('[A-Za-z]'), '#'),
       filter: {'#': RegExp('[0-9]')},
       type: MaskAutoCompletionType.eager,
     );
-    _setParams();
+
+    if (widget.initialValue != null) {
+      _setText(widget.initialValue!);
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        _key.currentState?.validate();
+      });
+    }
+    super.initState();
   }
 
   @override
-  void didUpdateWidget(ZetaDateInput oldWidget) {
-    super.didUpdateWidget(oldWidget);
-    _setParams();
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
   }
 
-  void _setParams() {
-    _size = widget.size ?? ZetaWidgetSize.large;
-    _hasError = widget.hasError;
+  void _onChange() {
+    if (_dateFormatter.getMaskedText().length == widget.dateFormat.length && (_key.currentState?.validate() ?? false)) {
+      widget.onChange?.call(_value);
+    }
+    setState(() {});
   }
 
-  void _onChanged() {
-    final value = _dateFormatter.getMaskedText().trim();
-    final date = DateFormat(widget.datePattern).tryParseStrict(value);
-    _invalidDate = value.isNotEmpty && date == null;
-    widget.onChanged?.call(date);
-    setState(() {});
+  void _setText(DateTime value) {
+    _controller.text = DateFormat(widget.dateFormat).format(value);
+    _dateFormatter.formatEditUpdate(TextEditingValue.empty, _controller.value);
   }
 
-  void _clear() {
-    _controller.clear();
+  Future<void> _pickDate() async {
+    final firstDate = widget.minDate ?? DateTime(0000);
+    final lastDate = widget.maxDate ?? DateTime(3000);
+    DateTime fallbackDate = DateTime.now();
+
+    if (fallbackDate.isBefore(firstDate) || fallbackDate.isAfter(lastDate)) {
+      fallbackDate = firstDate;
+    }
+
+    late final DateTime initialDate;
+
+    if (_value == null || (_value != null && _value!.isBefore(firstDate) || _value!.isAfter(lastDate))) {
+      initialDate = fallbackDate;
+    } else {
+      initialDate = _value!;
+    }
+
+    final result = await showDatePicker(
+      context: context,
+      firstDate: firstDate,
+      lastDate: lastDate,
+      initialDate: initialDate,
+      builder: (BuildContext context, Widget? child) {
+        return Theme(
+          data: Theme.of(context).copyWith(
+            datePickerTheme: const DatePickerThemeData(
```

Comment:
personal preference: i think this divider is too dark.
also it is really rounded, should / can this have a sharp variant?
![image](https://github.com/ZebraDevs/zeta_flutter/assets/23633655/148c5014-c551-4a74-aa10-dbc358b71b5d)

Reply:
yep will do. Do you want me to reduce the border radius on this when the component is rounded?

Reply:
yeah i think so